### PR TITLE
add multiline support for private key detector

### PR DESCRIPTION
**Pull request by maxamel**
This PR adds better support for private key plugin by working on the entire file instead of line-by-line

### CODE REVIEW

**Review by YaaraVerner — APPROVED** [review]

File: detect_secrets/plugins/private_key.py [hunk 1]
```diff
@@ -58,6 +64,39 @@ class PrivateKeyDetector(RegexBasedDetector):
...
+            filename: str,
+            line: str,
+            line_number: int = 0,
+            context: Optional[CodeSnippet] = None,
+            raw_context: Optional[CodeSnippet] = None,
+            **kwargs: Any
+    ) -> Set[PotentialSecret]:
+        output: Set[PotentialSecret] = set()
+
+        output.update(
```
Comment (bo156) [inline detect_secrets/plugins/private_key.py:81]: why do you need this part if you already read the entire file in the next part?

Comment (maxamel) [reply detect_secrets/plugins/private_key.py:81]: In some cases for private keys the line already contains a violation, and we don't need the file reading

Comment (maxamel) [reply detect_secrets/plugins/private_key.py:81]: It is similar to how our multiline detectors work today

**Review by bo156 — APPROVED** [review]

File: detect_secrets/plugins/private_key.py [hunk 2]
```diff
@@ -39,12 +44,13 @@ class PrivateKeyDetector(RegexBasedDetector):
     """
 
     secret_type = 'Private Key'
+    MAX_FILE_SIZE: int = 4 * 1024
```
Comment (nimrodkor) [inline detect_secrets/plugins/private_key.py:47]: @maxamel Where is this const used?

Comment (maxamel) [reply detect_secrets/plugins/private_key.py]: @nimrodkor  right, I wanted to add file size limitation before reading file, I added it now, please re-review

**Pull request merged**